### PR TITLE
Avoid race condition in vptr lookup

### LINKED ISSUE #4544: A ThreadSanitizer warning in the `eth` command
On version 3d6eb1fb26ea5c93c66bb380070f5347ec78974f, `eth/eth` shows the following warning

```
==================
WARNING: ThreadSanitizer: data race on vptr (ctor/dtor vs virtual call) (pid=11157)
  Write of size 8 at 0x7ba400000000 by main thread:
    #0 EthashClient /home/yh/src/cpp-ethereum/libethashseal/EthashClient.cpp:57 (discriminator 1) (eth+0x774982)
    #1 WebThreeDirect /home/yh/src/cpp-ethereum/libwebthree/WebThree.cpp:59 (discriminator 1) (eth+0x7ed672)
    #2 main /home/yh/src/cpp-ethereum/eth/main.cpp:929 (discriminator 1) (eth+0x502e7f)

  Previous read of size 8 at 0x7ba400000000 by thread T5:
    #0 dev::eth::Client::rejigSealing() /home/yh/src/cpp-ethereum/libethereum/Client.cpp:580 (eth+0x66b592)
    #1 dev::eth::Client::doWork(bool) /home/yh/src/cpp-ethereum/libethereum/Client.cpp:669 (eth+0x667c56)
    #2 dev::eth::Client::doWork() /home/yh/src/cpp-ethereum/libethereum/Client.h:237 (eth+0x67537e)
    #3 non-virtual thunk to dev::eth::Client::doWork() /home/yh/src/cpp-ethereum/libdevcore/../libethereum/Client.h:? (eth+0x6753c0)
    #4 dev::Worker::workLoop() /home/yh/src/cpp-ethereum/libdevcore/Worker.cpp:131 (eth+0x80ff30)
    #5 operator() /home/yh/src/cpp-ethereum/libdevcore/Worker.cpp:59 (eth+0x81045f)
    #6 void std::_Bind_simple<dev::Worker::startWorking()::$_0 ()>::_M_invoke<>(std::_Index_tuple<>) /usr/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0/functional:1390 (discriminator 2) (eth+0x8102d9)
    #7 std::_Bind_simple<dev::Worker::startWorking()::$_0 ()>::operator()() /usr/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0/functional:1380 (eth+0x810289)
    #8 std::thread::_State_impl<std::_Bind_simple<dev::Worker::startWorking()::$_0 ()> >::_M_run() /usr/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0/thread:197 (eth+0x81014d)
    #9 std::error_code::default_error_condition() const ??:? (libstdc++.so.6+0xbb83e)

  Location is heap block of size 20448 at 0x7ba400000000 allocated by main thread:
    #0 operator new(unsigned long) ??:? (eth+0x4edd73)
    #1 WebThreeDirect /home/yh/src/cpp-ethereum/libwebthree/WebThree.cpp:59 (eth+0x7ed609)
    #2 main /home/yh/src/cpp-ethereum/eth/main.cpp:929 (discriminator 1) (eth+0x502e7f)

  Thread T5 'eth' (tid=11440, running) created by main thread at:
    #0 pthread_create ??:? (eth+0x462346)
    #1 std::thread::_M_start_thread(std::unique_ptr<std::thread::_State, std::default_delete<std::thread::_State> >, void (*)()) ??:? (libstdc++.so.6+0xbbb54)
    #2 dev::Worker::startWorking() /home/yh/src/cpp-ethereum/libdevcore/Worker.cpp:44 (discriminator 1) (eth+0x80f900)
    #3 dev::eth::Client::init(dev::p2p::Host*, boost::filesystem::path const&, dev::WithExisting, boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<256u, 256u, (boost::multiprecision::cpp_integer_type)0, (boost::multiprecision::cpp_int_check_type)0, void>, (boost::multiprecision::expression_template_option)0>) /home/yh/src/cpp-ethereum/libethereum/Client.cpp:126 (eth+0x6674bb)
    #4 Client /home/yh/src/cpp-ethereum/libethereum/Client.cpp:78 (eth+0x666e49)
    #5 EthashClient /home/yh/src/cpp-ethereum/libethashseal/EthashClient.cpp:56 (eth+0x774968)
    #6 WebThreeDirect /home/yh/src/cpp-ethereum/libwebthree/WebThree.cpp:59 (discriminator 1) (eth+0x7ed672)
    #7 main /home/yh/src/cpp-ethereum/eth/main.cpp:929 (discriminator 1) (eth+0x502e7f)

SUMMARY: ThreadSanitizer: data race on vptr (ctor/dtor vs virtual call) /home/yh/src/cpp-ethereum/libethashseal/EthashClient.cpp:57 (discriminator 1) in EthashClient
==================
```

### PR DESCRIPTION
**Pull request by pirapira**
This PR removes some race conditions ThreadSanitizer reported during `eth/eth` runs.

Fixes #4544

### CODE REVIEW

File: libdevcore/Worker.cpp [hunk 1]
```diff
@@ -84,10 +87,6 @@ void Worker::startWorking()
 		}));
 //		cnote << "Spawning" << m_name;
 	}
-
-	DEV_TIMED_ABOVE("Start worker", 100)
-		while (m_state == WorkerState::Starting)
-			m_state_notifier.wait(l);
```
Comment (pirapira) [inline libdevcore/Worker.cpp]: These lines should not be removed.

File: libethereum/Client.h [hunk 2]
```diff
@@ -83,6 +83,7 @@ class Client: public ClientBase, protected Worker
 		int _networkID,
 		p2p::Host* _host,
 		std::shared_ptr<GasPricer> _gpForAdoption,
+		bool _final, // true when no derived object is to be constructed!
```
Comment (pirapira) [inline libethereum/Client.h]: :sob:

Is there any better way?

**Comment by gumb0** [comment]
Who calls `Worker::startWorking()` from constructor? I see ony `Client`.
We should fix it in the `Client` if it's the only problem

**Comment by pirapira** [comment]
OK, I'll try that.  Not much can be reused from here.

**Pull request closed without merging**